rllib/evaluation/sampler.py

```diff
@@ -51,14 +55,49 @@ class _PerfStats:
     def __init__(self):
         self.iters = 0
         self.env_wait_time = 0.0
-        self.processing_time = 0.0
+        self.pre_processing_time = 0.0
+        self.storing_samples_in_buffer = 0.0
         self.inference_time = 0.0
+        self.action_processing_time = 0.0
+        self.get_ma_train_batch = 0.0
+        self.postprocess_trajectories_so_far = 0.0
 
     def get(self):
+        # Mean multiplicator (1000 = ms -> sec).
+        factor = 1000 / self.iters
         return {
-            "mean_env_wait_ms": self.env_wait_time * 1000 / self.iters,
-            "mean_processing_ms": self.processing_time * 1000 / self.iters,
-            "mean_inference_ms": self.inference_time * 1000 / self.iters
+            # Waiting for environment (during poll).
+            "total_env_wait_s": self.env_wait_time,
+            "mean_env_wait_ms": self.env_wait_time * factor,
+            # Raw observation preprocessing.
+            "total_pre_processing_s": self.pre_processing_time,
+            "mean_pre_processing_ms": self.pre_processing_time * factor,
+            # Time spent storing/copying env/model data in the buffer.
+            "total_storing_samples_in_buffer_s":
+                self.storing_samples_in_buffer,
+            "mean_storing_samples_in_buffer_ms":
+                self.storing_samples_in_buffer * factor,
+            # Time spent during calls to get a ready train_batch.
+            "total_get_ma_train_batch_s": self.get_ma_train_batch,
+            "mean_get_ma_train_batch_ms": self.get_ma_train_batch * factor,
+            # Time spent during calls to "postprocess trajectories so far"
+            # methods.
+            "total_postprocess_trajectories_so_far_s":
+                self.postprocess_trajectories_so_far,
+            "mean_postprocess_trajectories_so_far_ms":
+                self.postprocess_trajectories_so_far * factor,
+            # Computing actions through policy.
+            "total_inference_ms": self.inference_time,
```

Comment:
Total isn't that useful if you have mean right?

Reply:
done